### PR TITLE
Zadanie 3 - TROSZKĘ spóżnione 

### PR DESCRIPTION
**Pull request by inamari-sus**

### CODE REVIEW

**Review by dubajstan — APPROVED** [review]
Bardzo dobrze zrobiona analiza CNN. Nie ma się do czego przyczepić.

**Comment by PilarToZiomal** [comment]
1. Ładowanie danych przez TensorFlow mimo wymogu korzystania z PyTorcha/torchvision — ujednolić proces do jednego frameworka.
2. Macierz pomyłek warto przedstawić w wersji znormalizowanej (%), co ułatwia porównania między klasami.
3. Brak wywołania `model.eval()` w `evaluate()` może zafałszować wyniki dla modeli z Dropout/BatchNorm.
4. BatchNorm został zdefiniowany, ale nie jest używany w `forward`.
5. Zachowaj kolejność warstw: **Conv → BatchNorm → ReLU → Pool**.
6. Modele są zapisywane pod tą samą ścieżką, co powoduje nadpisywanie poprzednich checkpointów.

Poza powyższymi uwagami kod jest bardzo dobrze napisany. Zaliczone.

**Comment by BlazewskiAdam** [comment]
Imo bardzo dobra praca. Proste i jasne opisy, miło się przegląda notebooka. Dobre i nieprzekombinowane eksperymenty z liczbą dodawanych warstw, kanałów, wartościami Batchnorm oraz Dropout. 

Myślę, że w 3 modelu wartoby dodać dropout, bo z tego co widze nie ma go w tej sieci
